main.c

```diff
@@ -396,18 +426,113 @@ void ResetIOP()
+
+Logical Port Refers to the Actual Port on the PS2. (Ports 1 and 2)
+
+The ASCII Reference Art Below Is taken From the PS2DEV Multitap Library Sample AND SOME INFO HAS BEEN ADDED/CORRECTED.
+(The Sample had D and C backwards)
+See https://github.com/ps2dev/ps2sdk/tree/master/ee/rpc/multitap For More Info
+//////////////////////////////////////////////////////////////////////
+						
+				     __________[ Port 1, Slot 3 ]
+                                    /    _____[ Port 1, Slot 2 ]
+ |------------|                     |   /
+ |            |                   |-------|
+ |            |                   | D   C |
+ |     PS2    |          |--------|       | <- Multi-tap
+ |            |          |        | A   B |
+ |            |          |        |-------|
+ |            |          |          |   \_____ [ Port 1, Slot 1 ]
+ |LogicalPort 2 |]---------|          \__________[ Port 1, Slot 0 ]
+ |            |
+ |LogicalPort 1 |]---[ Port 0, Slot 0 ]
+ |            |
+ |------------|
+//////////////////////////////////////////////////////////////////////
+Some More Notes
+
+ACCORING TO THE MULTITAP SAMPLE YOU MUST USE XMODULES IN ORDER TO USE THE MULTITAP!
+
+====Mtap Port Info====
+You DONT have to Open Mulitap Ports 1 & 2 if you are just looking to access the Memory Cards.
+mtapPortOpen(0); >> Memory Card Port 1 (Logical Controller Port 1)
+mtapPortOpen(1); >> Memory Card Port 2 (Logical Controller Port 2)
+mtapPortOpen(2); >> Memory Card Port 1 (Logical MC Port 1)
+mtapPortOpen(3); >> Memory Card Port 2 (Logical MC Port 2)
+======================
+
+====Port,Slot Info====
+ 1A: PORT = 0,SLOT = 0
+ 1B: PORT = 0,SLOT = 1
+ 1C: PORT = 0,SLOT = 2
+ 1D: PORT = 0,SLOT = 3
+ 
+ 2A: PORT = 1,SLOT = 0
+ 2B: PORT = 1,SLOT = 1
+ 2C: PORT = 1,SLOT = 2
+ 2D: PORT = 1,SLOT = 3
+====================== 
+ */
+{
+	closeMTAPports(); //In Case Multi-tap(s) are Connected or removed After the App has been started.
+	int rv;
+	mtapPortOpen(2);
+	mtapPortOpen(3);
+	scr_printf("Multi-tap Status: \n");
+	//Checks For Mtap Connection on Physical Memory Card Slot 1
+	rv = mtapGetConnection(2); // Checks MTAP port 2 (Memory Card Port 1) For MTAP Connection
+	
+	if(rv == 1)
+        scr_printf("Memory Card Port 1: Multi-tap Detected! \n");
+    else
+    {
+        scr_printf("Memory Card Port 1: Multi-tap is Not Connected. \n");
+        mtapPortClose(2); //Closes The Multitap Port if The Multitap Is Not Present
+    }
+	
+	
+	//Checks For Mtap Connection on Physical Memory Card Slot 2
+	rv = mtapGetConnection(3); // Checks MTAP port 3 (Memory Card Port 2) For MTAP Connection
+    
+	if(rv == 1)
+        scr_printf("Memory Card Port 2: Multi-tap Detected! \n");
+    else
+    {
+        scr_printf("Memory Card Port 2: Multi-tap is Not Connected. \n");
+        mtapPortClose(3);
+	}
+}
+// Closes MTAP Ports 2 and 3
+void closeMTAPports()
+{
+	mtapPortClose(2);
+	mtapPortClose(3);
+}
 void gotoOSDSYS()
```

Comment:
This should just be 1 function instead of 2.
Add a parameter int failure or something and if it was a failure pass in a 1. Make an error message and if failure ==1 print out the message otherwise print going back to osd